processing/src/main/java/io/druid/query/filter/BooleanFilter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.filter;
+
+import java.util.List;
+
+public interface BooleanFilter extends Filter
```

Comment:
I kept this after making the `flatten` modification to return new filters, but I'm leaning towards keeping this, since And/OrFilter didn't have a `getFilters()` method prior to this. It also lets me do (instance of BooleanFilter) instead of (instance of AndFilter || instance of OrFilter)